e2e/testenv/infra/cmd/setup.go

```diff
@@ -13,13 +13,38 @@ import (
 	"sync"
 	"time"
 
-	"github.com/spf13/cobra"
-	"helm.sh/helm/v3/pkg/registry"
-
 	"github.com/rancher/fleet/e2e/testenv"
 	"github.com/rancher/fleet/e2e/testenv/kubectl"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/pkg/registry"
 )
 
+var timeoutDuration = 10 * time.Minute // default timeout duration
+
+func eventually(f func() (string, error)) string {
```

Comment:
Liking this, wrapping anything that can fail on first attempts into this, as you did, makes more sense than the initial approach. Thanks for this!